src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/ApplicationAssembliesProvider.cs

```diff
@@ -151,30 +154,46 @@ private AssemblyItem GetAssemblyItem(Assembly assembly)
         // Internal for unit testing
         internal static IEnumerable<RuntimeLibrary> GetCandidateLibraries(DependencyContext dependencyContext)
         {
-            var candidatesResolver = new CandidateResolver(dependencyContext.RuntimeLibraries, ReferenceAssemblies);
-            return candidatesResolver.GetCandidates();
+            // When using Microsoft.AspNetCore.App \ Microsoft.AspNetCore.All shared runtimes, entries in the RuntimeLibraries
+            // get "erased" and it is no longer accurate to query to determine a library's dependency closue.
+            // We'll use CompileLibraries to calculate the dependency graph and runtime library to resolve assemblies to inspect.
+            var candidatesResolver = new CandidateResolver(dependencyContext.CompileLibraries, ReferenceAssemblies);
```

Comment:
Verified that this addresses the testing issue if the preview2 binary is replaced with a local build containing this change.